Quick-Start [DA]/MVP/MVP+Makefile.md

```diff
@@ -1 +1,894 @@
+-->
+
+For more information on the Makefiles, you can refer to the [PostgreSQL documentation on makefile](https://www.postgresql.org/docs/15/extend-pgxs.html).
+
+#### Tip: 
+```makefile
+EXTENSION = my_extension
+DATA = my_extension--1.0.0.sql my_extension--1.1.0.sql
+```
+
+- By including multiple SQL script files in the `DATA` variable of the `Makefile`, the user can ensure that different versions of the extension can have different sets of objects or functionality. 
+- Each SQL script file represents a specific version of the extension and can contain the necessary SQL commands for the desired changes.
+- This allows the user to easily manage and control the installation of specific features or modifications in different versions of your extension.
+
+### Step 5: Build and Install the Extension
+- To build and install the extension, run the following commands:
+
+```bash
+$ make
+$ make install
+```
+
+For more information on the installation procedures, you can refer to the [PostgreSQL documentation on Installing the Files](https://www.postgresql.org/docs/current/install-procedure.html#INSTALL).
+
+> NOTE: make: Nothing to be done for 'all'
+
+> Makefile doesn't have any MODULE or other binary to build. You can go directly to the make install step
+
+- The "make" command compiles the extension code using the Makefile
+- The "make install" command copies the necessary files to the appropriate PostgreSQL directories.
+
+### Step 6: Create the Extension in PostgreSQL:
+- Connect to your PostgreSQL database and execute the following SQL statement to create the extension:
+
+```sql
+CREATE EXTENSION IF NOT EXISTS my_extension;
+-- Output: CREATE EXTENSION
+```
+- The CREATE EXTENSION statement is executed in PostgreSQL to create the extension.
+
+For more information on the `CREATE EXTENSION` command, you can refer to the [PostgreSQL documentation on `CREATE EXTENSION`](https://www.postgresql.org/docs/current/sql-createextension.html).
+
+
+## Implementation of the Extension:
+Once you have successfully implemented the `my_extension` in your PostgreSQL database, you can use it as follows:
+
+### 1. Query the `my_table` table:
+   - You can query the `my_table` table like any other table in your database. For example, you can retrieve all rows from the table using the following SQL query:
+   ```sql
+   SELECT * FROM my_table;
+   ```
+#### Output:
+```
+id | name
+----+------
+(0 rows)
+```
+### 2. Use the add function to perform addition:
+  - The `add` function takes two integer arguments and returns their sum
+  ```sql
+  SELECT add(2, 3);
+  ```
+#### Output:
+```
+ add
+-----
+   5
+(1 row)
+```
+
+## Regression Testing:
+
+### Step 1: Create the Extension SubDirectory for Regression SQL File:
+- Create a subdirectory for your regression file. Name it `sql`
+
+```bash
+$ mkdir sql
+$ cd sql
+```
+-   Create a new file called `my_extension--regress.sql`.
```

Comment:
I recommend using a single dash, here; double dashes are used in extension file names because they have special meaning, separating names from versions.

Reply:
Sure i will make the changes.

Reply:
I don't understand the comment. You don't need the word `regression` in the file name.

Reply:
Okay, i miss understood. I just was using a particular extension as an example. He used --regress.sql, so i followed suit.